README.md
# HybridOS
HybridOS has now run its course, and we have moved onto working on MonoOS.
We've decided to set the program free and it will likely no longer be maintained.
This is the code for the original HybridOS 2, programmed by me and @Mantaraix in Python, for you to do whatever the heck you want with.
Credit to most code written in this program goes to StackOverflow.


HybridOStxtFiles/Changelog.txt



╔═██  ╔═██          ╔═██                 ╔██      ╔═██ ╔═██████  ╔═██████
║ ██  ║ ██          ║ ██                 ╚═╝      ║ ██╔╝██══╗ ██╔╝██══╗ ██
║ ██  ║ ██╔═██  ╔═██║ ███████  ╔═██████  ╔██ ╔═███████║ ██  ║ ██║ ██  ╚══╝
║ ████████║ ██  ║ ██║ ██══╗ ██╔╝██══╗ ██ ║██╔╝██══╗ ██║ ██  ║ ██╚╗ ██████ 
║ ██══╗ ██║ ██  ║ ██║ ██  ║ ██║ ██  ╚══╝ ║██║ ██  ║ ██║ ██  ║ ██ ╚════╗ ██
║ ██  ║ ██║ ██  ║ ██║ ██  ║ ██║ ██       ║██║ ██  ║ ██║ ██  ║ ██╔═██  ║ ██
║ ██  ║ ██╚╗ ███████║ ███████╝║ ██       ║██╚╗ ███████╚╗ ██████╝╚╗ ██████╝
╚══╝  ╚══╝ ╚════╗ ██╚═══════╝ ╚══╝       ╚═╝ ╚═══════╝ ╚══════╝  ╚══════╝ 
           ╔═██ ║ ██                                                     
           ║ ██████╝                                                      
           ╚══════╝                                                                                      


All changes for the Vanilla HybridOS Indev Main.py file listed below from Version 2.8.5 Onwards!

═══════════════════════════════════════════════════════════════════════════

Version 2.8.5: 

- @changes
  print("6/21/2019 New loading screen and with more aesthetic code!")
  

═══════════════════════════════════════════════════════════════════════════


Version 2.8.7:

- @changes
  print("6/21/2019 - More commands and the Time Update!")
  

═══════════════════════════════════════════════════════════════════════════

Version 2.9.5: 

- @changes
  print("6/21/2019 - New shutdown system, game system in development. ")
  
═══════════════════════════════════════════════════════════════════════════

Version 2.9.7:

-@changes
  print("6/22/2019 - Updated game system. ")


═══════════════════════════════════════════════════════════════════════════

Version 2.10.1:

-@changes
  print("6/22/2019 - More colour and the system cares less about capitalization. Further updated game system.") 

═══════════════════════════════════════════════════════════════════════════

Version 2.10.7:

-@changes
  print("6/22/2019 - Updated HybridOS Startup logo, new  Guess the Number game in the GameRoom and new root system.") 

═══════════════════════════════════════════════════════════════════════════

Version 2.11.1:

-@changes
  print("6/23/2019 - Fixed HybridOS Startup logo, new Hangman game in the GameRoom.") 

═══════════════════════════════════════════════════════════════════════════

Version 2.11.2:

-@changes
  print("6/23/2019 - HybridOS cares even less about capitalization, and Happy 500th line of code!") 

═══════════════════════════════════════════════════════════════════════════

Version 2.11.3:

-@changes
  print("6/24/2019 - Root now appears in commands, and new guest account!") 

═══════════════════════════════════════════════════════════════════════════

Version 2.11.4:

-@changes
  print("6/24/2019 - Guests now cannot access root and root is now encrypted and has been made more secure.") 

═══════════════════════════════════════════════════════════════════════════

Version 2.11.6 Cyborg:

-@changes
  print("6/25/2019 - OS version now has a name - Cyborg. Subject to change.) 
	print("6/25/2019 - New logoff system. Also changed gameroom commands to yellow. Changed @Exitgr to @Grexit.) 

═══════════════════════════════════════════════════════════════════════════

Version 2.11.8 Cyborg:

-@changes
  print("6/30/2019 - New 400-Line Noughts and Crosses!) 
	

═══════════════════════════════════════════════════════════════════════════

Version 2.11.9 Cyborg:

-@changes
  print("Updated root and updated HybridOS files including the License.") 
	
═══════════════════════════════════════════════════════════════════════════

Version 2.11.10 Cyborg:

-@changes
  print("Updated Hangman game. Updated UI a bit including loading and logon screen.") 
	
═══════════════════════════════════════════════════════════════════════════

Version 2.11.11 Cyborg:

-@changes
  print("7/13/2019 L - New official HybridOS distros (Now Home, Lite, Superlite.) as well as some logon screen slight changes.") 
	
═══════════════════════════════════════════════════════════════════════════

Version 2.11.12 Cyborg:

-@changes
  print("7/13/2019 L - New credits page.") 
	
═══════════════════════════════════════════════════════════════════════════

Version 2.13.0 Cyborg:

-@changes
  print("7/13/2019 L - New Ideas and Bugs document.") 
	print("7/13/2019 L - Fixed bug where hangman head was misaligned with the rope.") 
	print("7/14/2019 L - Changed name from Home to Indev.") 
	print("7/14/2019 L - Added 'GMT' to the time in the info section.") 
	print("7/15/2019 L - Fixed bug where hangman would only start hanging at guess 5.") 
	print("7/15/2019 L - Fixed bug where it would say 'Command not recognized' after every item/app.") 
	print("7/15/2019 L - Slight changes to gameroom versatility.") 
	print("7/15/2019 L - Added one more person to the credits page.") 
-@notes
	print("7/15/2019 L - This deserves a 2-MC (major change) jump!") 

═══════════════════════════════════════════════════════════════════════════

Version 2.13.2 Cyborg:

-@changes
  print("7/16/2019 L - Near infinite versitality in all commands, all command engines no longer care about capitalization.") 

═══════════════════════════════════════════════════════════════════════════

Version 2.13.4 Cyborg:

-@changes
  print("8/8/2019 L - Alternating colours on command screens, more colour. Changed some colours.") 

═══════════════════════════════════════════════════════════════════════════

Version 2.13.7 Cyborg:

-@changes
  print("8/10/2019 L - Added error screen and startup animation.") 
-@notes
	print("8/10/2019 L - So basically I put the whole program in a try-except loop.") 

═══════════════════════════════════════════════════════════════════════════

Version 2.13.9 Cyborg:

-@changes
  print("8/11/2019 L - Added the ability to show various text files in HybridOS. Also added the ability to show webpage code.") 

═══════════════════════════════════════════════════════════════════════════

Version 2.13.11 Cyborg:

-@changes

test.py
import time
# Imports #
# AppCode #
print("Initializing HybridTime...")
time.sleep(1)





startupAnim.py
def aaa():
	import os
	from os import system
	import time
	from time import sleep
	from colorama import Fore

	print(Fore.MAGENTA + "\n\n          ╔═██  ╔═██          ╔═██                 ╔██      ╔═██ ╔═██████  ╔═██████")
	print(Fore.MAGENTA + "          ║ ██  ║ ██          ║ ██                 ╚═╝      ║ ██╔╝██══╗ ██╔╝██══╗ ██")
	print(Fore.BLUE + "          ║ ██  ║ ██╔═██  ╔═██║ ███████  ╔═██████  ╔██ ╔═███████║ ██  ║ ██║ ██  ╚══╝")
	print(Fore.BLUE + "          ║ ████████║ ██  ║ ██║ ██══╗ ██╔╝██══╗ ██ ║██╔╝██══╗ ██║ ██  ║ ██╚╗ ██████ ")
	print(Fore.CYAN + "          ║ ██══╗ ██║ ██  ║ ██║ ██  ║ ██║ ██  ╚══╝ ║██║ ██  ║ ██║ ██  ║ ██ ╚════╗ ██")
	print(Fore.GREEN + "          ║ ██  ║ ██║ ██  ║ ██║ ██  ║ ██║ ██       ║██║ ██  ║ ██║ ██  ║ ██╔═██  ║ ██")
	print(Fore.GREEN + "          ║ ██  ║ ██╚╗ ███████║ ███████╝║ ██       ║██╚╗ ███████╚╗ ██████╝╚╗ ██████╝")
	print(Fore.YELLOW + "          ╚══╝  ╚══╝ ╚════╗ ██╚═══════╝ ╚══╝       ╚═╝ ╚═══════╝ ╚══════╝  ╚══════╝  ")
	print(Fore.YELLOW + "                     ╔═██ ║ ██                                                      ")
	print(Fore.RED + "                     ║ ██████╝                                              ")
	print(Fore.RED + "                     ╚══════╝                   \n")

	sleep(0.8)
	system("clear")

	print(Fore.RED + "\n\n          ╔═██  ╔═██          ╔═██                 ╔██      ╔═██ ╔═██████  ╔═██████")
	print(Fore.RED + "          ║ ██  ║ ██          ║ ██                 ╚═╝      ║ ██╔╝██══╗ ██╔╝██══╗ ██")
	print(Fore.YELLOW + "          ║ ██  ║ ██╔═██  ╔═██║ ███████  ╔═██████  ╔██ ╔═███████║ ██  ║ ██║ ██  ╚══╝")
	print(Fore.YELLOW + "          ║ ████████║ ██  ║ ██║ ██══╗ ██╔╝██══╗ ██ ║██╔╝██══╗ ██║ ██  ║ ██╚╗ ██████ ")
	print(Fore.GREEN + "          ║ ██══╗ ██║ ██  ║ ██║ ██  ║ ██║ ██  ╚══╝ ║██║ ██  ║ ██║ ██  ║ ██ ╚════╗ ██")
	print(Fore.GREEN + "          ║ ██  ║ ██║ ██  ║ ██║ ██  ║ ██║ ██       ║██║ ██  ║ ██║ ██  ║ ██╔═██  ║ ██")
	print(Fore.CYAN + "          ║ ██  ║ ██╚╗ ███████║ ███████╝║ ██       ║██╚╗ ███████╚╗ ██████╝╚╗ ██████╝")
	print(Fore.BLUE + "          ╚══╝  ╚══╝ ╚════╗ ██╚═══════╝ ╚══╝       ╚═╝ ╚═══════╝ ╚══════╝  ╚══════╝  ")
	print(Fore.BLUE + "                     ╔═██ ║ ██                                                      ")
	print(Fore.MAGENTA + "                     ║ ██████╝                                              ")
	print(Fore.MAGENTA + "                     ╚══════╝                   \n")

	sleep(0.8)
	system("clear")

OtherFiles/blakbird.py

# now this is REALLY old code, most of this is copied from the internet
# coded by lewin pre-2018

import urllib.request
import time
import random
import string
from random import randint

print("BlakBird Allium v1.1.0. Booting Up.\n\nCopyright © VenimueCelestia Technologies Incorporated. All rights reserved.")

while True:
    username = input("\nWelcome to BlakBird! Please type your username:\n\n")

    username1 = "Lewin"
    username2 = "Username"
    
    if username == username1:
        print ("\nWelcome back, Creator! Please enter your password.")
        break
    elif username == username2:
        print ("\nHi there! Please enter your password.")
        break
    else:
        print ("\nThere is no user with this name. Try with caps?")
        
while True:
    password = input("\nPassword:\n\n")
        
    password1 = "Access"
    password2 = "Password"

    if password == password1 and username == username1:
        print ("\nHi there, Lewin!")
        break
    elif password == password2 and username == username2:
        print ("\nWelcome to the system, Epic!")
        break
    else:
        print ("\nAccess denied. All passwords are case-sensitive.")
        
print("Choices:\n \nShut down\nPrint notes\nInternet (ALPHA)\nDice Roll\nCalculator\n\nGAMES:\nPlay FizzBuzz\nMagic 8 ball\nGuess The Number\nComputer Quiz\nText Evolution\nHangman\nEscape the Cavern\nDroids")

while True:
    choice = input("\nWhat do you want to do?\n\n")

    if choice == "shut down" or choice == "shutdown":
        print("Shutting Down.")
        break

    elif choice == "print notes" or choice == "notes":
        if username == username1:
            print("\nHello\nThese are notes\nSub to pewdiepie\nNotes are cool\nthis operating system is cool\n\nTo Do:\nStop the system from crashing whenever I type a non-number into the quiz/calculator\n\nbye")
        else:
            print("\nEPIC EPIC EPIX SPACESHIPPPPPPPPPPPPPPPPPPPPPPPPPPPZZZZ R COOL HOW BOUT U")
    elif choice == "play fizzbuzz" or choice == "fizzbuzz":
        print('FizzBuzz is a game where you replace the multiples of three with "Fizz", the multiples of five with "Buzz" and the multiples of both with "FizzBuzz."\n')
        for i in range(1,101): print("Fizz"*(i%3==0) + "Buzz"*(i%5==0) or i, end='\n')

    elif choice == "magic 8 ball" or choice == "8 ball":
        while True:
            question = input("Ask the Magic 8 ball a question: (Press enter to quit.) ")
            answers = random.randint(1,8)
            if question == "":
                break
            elif answers == 1:
                print("It is certain")
            elif answers == 2:
                print ("Outlook good")
            elif answers == 3:
                print ("You may rely on it")
            elif answers == 4:
                print ("Ask again later")
            elif answers == 5:
                print ("Concentrate and ask again")
            elif answers == 6:
                print ("Reply hazy, try again")
            elif answers == 7:
                print ("My reply is no")
            elif answers == 8:
                print ("My sources say no")
                
    elif choice == "internet" or choice == "net":
        webUrl  = urllib.request.urlopen('https://www.aol.co.uk/')
        print ("result code: " + str(webUrl.getcode()))
        data = webUrl.read()
        print (data)
        
    elif choice == "guess the number":
        n = random.randint(1, 99)
        guess = int(input("Enter an integer from 1 to 99: "))
        while n != "guess":
            print
            if guess < n:
                print("Higher!")
                guess = int(input("Enter an integer from 1 to 99: "))
            elif guess > n:
                print("Lower!")
                guess = int(input("Enter an integer from 1 to 99: "))
            else:
                print("you guessed it!")
                break
            
    elif choice == "dice roll" or choice == "dice":
        min = 1
        max = 6
        roll_again = "yes"
        while roll_again == "yes" or roll_again == "y":
            print("Rolling the dices...")
            print("The values are....")
            print(random.randint(min, max))
            print(random.randint(min, max))
            roll_again = input("Roll the dices again? Yes or no?\n\n")
            
    elif choice == "computer quiz" or choice == "quiz":
        my_dict =   {
                        "Base-2 number system" : "binary",
                        "Number system that uses the characters 0-F" : "hexidecimal",
                        "7-bit text encoding standard" : "ascii",
                        "16-bit text encoding standard" : "unicode",
                        "A number that is bigger than the maximum number that can be stored" : "overflow",
                        "8 bits" : "byte",
                        "1024 bytes" : "kilobyte",
                        "Picture Element. The smallest component of a bitmapped image" : "pixel",
                        "A continuously changing wave, such as natural sound" : "analogue",
                        "the number of times per second that a wave is measured" : "sample rate",
                        "A binary representation of a program" : "machine code"
                    }
        print("Welcome to... The computer Quiz!!!")
        print("==================================")
        playing = True
        while playing == True:
            score = 0
            num = int(input("\nHow many questions would you like?\n\n"))
            for i in range(num):
                question = (random.choice( list(my_dict.keys())))
                answer = my_dict[question]
                print("\nQuestion " + str(i+1) )
                print(question  + "?")
                guess = input("> ")
                if guess.lower() == answer.lower():
                    print("Correct!")
                    score += 1
                else:
                    print("Nope!") 
            print("\nYour final score was " + str(score))
            again = input("Enter any key to play again, or 'q' to quit.\n\n")
            if again.lower() == 'q':
                playing = False
                
    elif choice == "text evolution" or choice == "text":
        possibleCharacters = string.ascii_lowercase + string.digits + string.ascii_uppercase + ' .,!?;:'
        target = input("Enter your target text: ")
        attemptThis = ''.join(random.choice(possibleCharacters) for i in range(len(target)))
        attemptNext = ''
        completed = False
        generation = 0
        while completed == False:
            print(attemptThis)
            attemptNext = ''
            completed = True
            for i in range(len(target)):
                if attemptThis[i] != target[i]:
                    completed = False
                    attemptNext += random.choice(possibleCharacters)
                else:
                    attemptNext += target[i]
            generation += 1
            attemptThis = attemptNext
            time.sleep(0.035)
        print("Target matched! That took " + str(generation) + " generations!")
        
    elif choice == "calculator" or choice == "calc":
        print("1. Addition");
        print("2. Subtraction");
        print("3. Multiplication");
        print("4. Division");
        print("5. Exit");
        choice = int(input("Enter your choice: "));
        if (choice>=1 and choice<=4):
            print("Enter two numbers: ");
            num1 = int(input());
            num2 = int(input());
            if choice == 1:
                res = num1 + num2;
                print("Result = ", res);
            elif choice == 2:
                res = num1 - num2;
                print("Result = ", res);
            elif choice == 3:
                res = num1 * num2;
                print("Result = ", res);
            else:
                res = num1 / num2;
                print("Result = ", res);
        elif choice == 5:
            exit();
        else:
            print("Invalid input.");
            
    elif choice == "hangman":
        import random
        import time
        from time import sleep

        nesne="HANGMAN"
        simge="-"*50
        bos=""*5
        bilgi="""            You will earn 50 points if you guess a character correctly
                    and you will lose 25 points if your answer is wrong.
                    If you give 5 wrong answers you will lose."""
        print("{:^80}\n{:^80}\n{:^80}\n{:^80}\n{}".format(nesne,by,date,simge,bos))
        print(bilgi)
        print("{:^80}\n{:^80}\n{:^80}".format(bos,simge,bos))

        kelimeler = ["ability","about","above","absolute","accessible","accommodation",
                     "accounting","beautiful","bookstore","calculator","clever","engaged",
                     "engineer","enough","handsome","refrigerator","opposite","socks",
                     "interested","strawberry","backgammon","anniversary","confused",
                     "dangerous","entertainment","exhausted","impossible","overweight",
                     "temperature","vacation","scissors","accommodation","appointment",
                     "decrease","development","earthquake","environment","brand",
                     "environment","necessary","luggage","responsible","ambassador",
                     "circumstance","congratulate","frequent",]
        secim = random.choice(kelimeler)
        tablov=[]
        tablov.append("-"*len(secim))
        tablo=[]
        for i in tablov:
            for h in i:
                tablo.append(h)
        print("Our word has {} letters.".format(len(secim)),' '.join(tablo))
        print(""*7)
        c="""  _________
          |
          |"""
        c1="\n  O"
        c2="\n \|/"
        c3="\n  |"        
        c4="\n / \ "
        cson=""
        depo=""
        depo1=""
        a=len(depo)
        can=5
        puan=0
        olmazlar="+/ 1234567890*-_?.,"
        while True:
            if len(depo)==len(secim): 
                print("You won! Word: {} Your score: {}".format(secim,puan))
                print("Please press 'Enter' for quit..")
                break
            if can==0:
                print("\nYou lose...")
                sleep(0.9)
                print ("Your score: {}".format(puan))
                sleep(0.9)
                print ("The word that you couldnt answer: {}".format(secim))
                sleep(0.9)
                print("Please press 'Enter' for quit..")
                break

            x=input("\nPlease enter a letter: ")
            if 1<len(x):
                print ("\nYou can only enter 1 letter at a time.")
                continue
            if x in olmazlar:
                print("\nThis is not even a letter!")
                continue
            if x in depo:
                print("You have used this letter before!")
                continue
            if x in depo1:
                print("You have used this letter before!")
                continue
            if x in secim and not x in depo:
                puan+=50*secim.count(x)
                print("Letter {} is found {} times in our word!")
                #".format(x,secim.count(x)))
                sleep(0.7)
                for sayi, oge in enumerate(secim):
                    if 2 <=secim.count(oge) and x==oge:
                        depo+=x
                        a+=secim.count(oge)
                        if tablo[sayi]=="-": 
                            tablo[sayi]=x
                        print("In line {} .".format(sayi+1))
                        sleep(0.7)
                    if secim.count(oge)==1 and x==oge:
                        depo+=x
                        a+=secim.count(oge)
                        if tablo[sayi]=="-":
                            tablo[sayi]=x
                        print("In line {} .".format(sayi+1))
                        sleep(0.7)
                print("\nWord:",' '.join(tablo))
                sleep(0.7)

            else:
                depo1+=x
                can-=1
                puan-=25
                print("\nThis letter {} not in our word! {} guesses left.".format(x,can))
                sleep(0.7)
                print("\nWord:",' '.join(tablo))
                sleep(0.7)
                if can==4:
                    cson+=str(c)
                    print(cson)
                if can==3:
                    cson+=str(c1)
                    print(cson)      
                if can==2:          
                    cson+=str(c2)
                    print(cson)
                if can==1:
                    cson+=str(c3)
                    print(cson)
                if can==0:
                    cson+=str(c4)
                    print(cson)
        try:
            input()
        except SyntaxError:
            pass

